#536 ability to enable high-level commands after low-level commands

jpreiss opened the pull request:
Extensive discussion of the code design is in #536. This code is tested on real hardware. Confirmed that we can:

- Take off in high-level mode
- Fly using streaming setpoints (I tested `velocity_world` and `full_state`)
- Re-enable high-level mode
- Immediately land or do a goTo

I was not able to test manual override because I was not sure how. However, based on the design I am confident it will work.

krichardsson commented on src/modules/src/crtp_commander.c:
```diff
@@ -55,8 +103,19 @@ static void commanderCrtpCB(CRTPPacket* pk)
+        }
+      }
+      break;
+    }
```
Named constants makes it easier to understand what the cases are handling. Maybe something like 
```
switch (pk->channel) {
  case SET_SETPOINT_CHANNEL:
    // ...
    break;
  case META_COMMAND_CHANNEL:
    // ...
    break;
  default:
    // Do nothing
    break;
}
```

and a default for completeness.

jpreiss commented:
@krichardsson Thank you for the review! I added the channel `enum` and got rid of the `DEBUG_PRINT`s. 

Following the style of `crtp_commander_generic.c`, I also added comments in `crtp_commander.c` with a numbered "to-do" list for adding a new command, and changed the comments from `//` to `/* */` style.

krichardsson commented:
Merged. Thanks!
